[const] Move `CONF_ENABLED` to const.py

Gnuspice opened the pull request:
# What does this implement/fix?

Shifts definition of `CONF_ENABLED` from modbus_controller and wireguard components into const.py

Groundwork in preparation for a feature addition that would otherwise run afoul of "If a key appears in three or more components" rule.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code quality improvements to existing code or addition of tests
- [x] Other

**Related issue or feature (if applicable):**

- fixes N/A

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):**

- esphome/esphome-docs# N/A

## Test Environment

- [ ] ESP32
- [x] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx
- [ ] nRF52840

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
modbus_controller:
  - modbus_id: modbus_server
    [...]
    server_courtesy_response:
      enabled: true
      register_last_address: 0x0001
      register_value: 12345

binary_sensor:
  - platform: wireguard
    enabled:
      name: 'WireGuard Enabled'
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

Gnuspice committed:
[const] Move `CONF_ENABLED` to const.py

Gnuspice committed:
[const] Move `CONF_ENABLED` - fix linter complaint

clydebarrow commented:
> Hmm, pre-commit hooks didn't complain about this

The pre-commit hooks don't do everything that the CI checks do.

Gnuspice committed:
[const] Move `CONF_ENABLED` to components/const/__init__.py

Shifted from esphome/const.py as per review comments, as it's only used
inside components.

Gnuspice commented:
> I'd suggest moving to `components/const/__init__.py` instead. If the constant is only used in components, this means that the PR and any depending on it can be used via `external_components:` whereas any core changes can't be tested without cloning the PR branch.

Only used in components, so have made this change in the latest commit.
